File: src/test/java/dev/retrotv/crypt/twe/lea/LEACTRTest.java
package dev.retrotv.crypt.twe.lea;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import javax.crypto.spec.IvParameterSpec;
import java.security.Key;

import static org.junit.jupiter.api.Assertions.assertEquals;

class LEACTRTest {

    @Test
    @DisplayName("LEACTR-128 암복호화 테스트")
    void leactr128_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACTR lea = new LEACTR128();
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }

    @Test
    @DisplayName("LEACTR-192 암복호화 테스트")
    void leactr192_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACTR lea = new LEACTR192();
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }

    @Test
    @DisplayName("LEACTR-256 암복호화 테스트")
    void leactr256_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACTR lea = new LEACTR256();
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }
}

File: src/test/kotlin/dev/retrotv/crypto/twe/lea/LEACTRTest.kt
package dev.retrotv.crypto.twe.lea

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

internal class LEACTRTest {

    @Test
    @DisplayName("LEACTR-128 암복호화 테스트")
    fun leactr128_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACTR(128)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val params = ParamsWithIV(key.encoded, iv.iv)

        val encryptedData = lea.encrypt(message.toByteArray(), params)
        val originalMessage = String(lea.decrypt(encryptedData, params))
        Assertions.assertEquals(message, originalMessage)
    }

    @Test
    @DisplayName("LEACTR-192 암복호화 테스트")
    fun leactr192_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACTR(192)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val params = ParamsWithIV(key.encoded, iv.iv)

        val encryptedData = lea.encrypt(message.toByteArray(), params)
        val originalMessage = String(lea.decrypt(encryptedData, params))
        Assertions.assertEquals(message, originalMessage)
    }

    @Test
    @DisplayName("LEACTR-256 암복호화 테스트")
    fun leactr256_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACTR(256)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val params = ParamsWithIV(key.encoded, iv.iv)

        val encryptedData = lea.encrypt(message.toByteArray(), params)
        val originalMessage = String(lea.decrypt(encryptedData, params))
        Assertions.assertEquals(message, originalMessage)
    }
}
